Packet++/header/NtpLayer.h

```diff
@@ -0,0 +1,649 @@
+         */
+        int8_t getPrecision() const;
+
+        /**
+         * Set the value of precision
+         * @param[in] val Precision in log2 seconds
+         */
+        void setPrecision(int8_t val);
+
+        /**
+         * Get the value of precision in seconds
+         */
+        double getPrecisionInSecs() const;
+
+        /**
+         * Get the value of root delay in NTP short format
+         */
+        uint32_t getRootDelay() const;
+
+        /**
+         * Set the value of root delay
+         * @param[in] val Root delay in NTP short format
+         */
+        void setRootDelay(uint32_t val);
+
+        /**
+         * Get the value of root delay in seconds
+         */
+        double getRootDelayInSecs() const;
+
+        /**
+         * Set the value of root delay
+         * @param[in] val Root delay in seconds
+         */
+        void setRootDelayInSecs(double val);
+
+        /**
+         * Get the value of root dispersion in NTP short format
+         */
+        uint32_t getRootDispersion() const;
+
+        /**
+         * Set the value of root delay
+         * @param[in] val Root dispersion in NTP short format 
+         */
+        void setRootDispersion(uint32_t val);
+
+        /**
+         * Get the value of root dispersion in seconds
+         */
+        double getRootDispersionInSecs() const;
+
+        /**
+         * Set the value of root dispersion
+         * @param[in] val Root dispersion in seconds
+         */
+        void setRootDispersionInSecs(double val);
+
+        /**
+         * Get the value of reference identifier
+         */
+        uint32_t getReferenceIdentifier() const;
+
+        /**
+         * Set the value of reference identifier
+         * @param[in] val Value of the reference identifier, either NTPClockSource, IPv4 address or MD5 hash of first four octets of IPv6
+         */
+        void setReferenceIdentifier(uint32_t val);
+
+        /**
+         * Get the value of reference identifier as a string
+         * @return std::string String representation of NTP clock source if stratum is 1, IPv4 address or MD5 hash of first four octets of IPv6
+         */
+        std::string getReferenceIdentifierString() const;
+
+        /**
+         * Get the value of reference timestamp
+         * @return Value in NTP timestamp format 
+         */
+        uint64_t getReferenceTimestamp() const;
+
+        /**
+         * Set the value of reference timestamp
+         * @param[in] val Timestamp in NTP timestamp format
+         */
+        void setReferenceTimestamp(uint64_t val);
+
+        /**
+         * Get the value of reference timestamp
+         * @return Value in seconds from Unix Epoch (1 Jan 1970)
+         */
+        double getReferenceTimestampInSecs() const;
+
+        /**
+         * Set the value of reference timestamp
+         * @param[in] val Value in seconds from Unix Epoch (1 Jan 1970)
+         */
+        void setReferenceTimestampInSecs(double val);
```

Comment:
> @seladb Let's continue from here to follow up easily. Searching from reviews a bit confusing
> 
> Updated regarding your reviews. Except this,
> 
> > We don't have tests for all the `set...InSecs` methods (`setReferenceTimestampInSecs`, `setReceiveTimestampInSecs`, `setRootDelayInSecs`, `setRootDispersionInSecs`). Maybe the easiest way would be to add to `NtpCraftingTests` crafting of a NTPv3 packet which uses these methods 🤔
> 
> _Originally posted by @seladb in [#788 (comment)](https://github.com/seladb/PcapPlusPlus/pull/788#discussion_r795163539)_
> 
> And my comment,
> 
> > @seladb set....InSecs methods have some fundamental problem. Since they are using floating point, same as get...InSecs functions their result may or may not equal to provided value so the tests can fail. Because of this i lower the precision of this to 4 digits too. Because it can change sometimes from platform to platform too.
> > Because of the same problem NTP timestamp format interprets integer and fraction parts separately. I can replace these double function calls with structs,
> > struct X
> > {
> > uint32_t integer,
> > float fraction
> > }
> > but this means they will equal to plain set....() methods because it is actually NTP Timestamp format.
> > Do you have any suggestions for to test these methods?

If I understand correctly the value may change from platform to platform because the accuracy of `double` varies between platforms, is that correct?
In that case, can we use an input value that is low enough and will be supported on all platforms?